modules/fedimint-ln-client/src/lib.rs

```diff
@@ -190,12 +229,120 @@ impl LightningClientExt for Client {
         Ok(operation_id)
     }
 
+    async fn create_bolt11_invoice_and_receive(
+        &self,
+        amount: Amount,
+        description: String,
+        expiry_time: Option<u64>,
+        gateway: LightningGateway,
+    ) -> anyhow::Result<OperationId> {
+        let (ln_client_id, ln_client) = ln_client(self);
+        let network = self.get_network();
+        let (invoice, payment_hash, preimage, keypair) = ln_client
+            .generate_unconfirmed_invoice(
+                amount,
+                description,
+                rand::rngs::OsRng,
+                expiry_time,
+                gateway,
+                network,
+            )
+            .await?;
+        let operation_id = invoice.payment_hash().into_inner();
+        let output = ln_client.create_offer_output(
+            operation_id,
+            amount,
+            invoice.clone(),
+            keypair,
+            payment_hash,
+            preimage,
+            expiry_time,
+        )?;
```

Comment:
Why are `generate_unconfirmed_invoice` and `create_offer_output` two functions? Would we ever want to generate an unconfirmed invoice without also creating the output?

Reply:
This is mainly an artifact of copying the existing code, I will refactor so its clearer. You are correct, two functions are not needed.

Reply:
Refactored into one function